app/models/minion.rb

```diff
@@ -22,7 +25,7 @@ class NonExistingMinion < StandardError; end
   #   )
   def self.assign_roles!(roles: {}, default_role: :minion)
     requested_master = roles[:master]
-    requested_minions = roles[:minion]
+    requested_minions = roles[:minion] || (Minion.unassigned_role.pluck(:id) - [requested_master])
```

Comment:
[MINOR] After this addition, the check on line 34 below is no longer needed I think.